src/inventorymanager.cpp

```diff
@@ -413,9 +422,10 @@ void IMoveAction::apply(InventoryManager *mgr, ServerActiveObject *player, IGame
 	/* Modify count according to collected data */
 	count = src_item.count;
 	if (src_can_take_count != -1 && count > src_can_take_count)
-		count = src_can_take_count;
+		count = std::min((int)count, src_can_take_count);
```

Comment:
Probably unneeded?

Reply:
Not needed because of the check `count > src_can_take_count` above.